garak/generators/huggingface.py

```diff
@@ -96,6 +103,15 @@ def _gather_hf_params(self, hf_constructor: Callable):
                     continue
                 args[k] = params[k]
 
+        if (
+            not "device_map" in args
+            and "device_map" in params_to_process
+            and "device" in params_to_process
+            and "device" in args
+        ):
+            del args["device"]
+            args["device_map"] = self.device
+
```

Comment:
we're going to want to factor this kind of logic out and up somewhere one day. not in this pr though.